cmd/zetatool/filterdeposit/btc.go

```diff
@@ -0,0 +1,188 @@
+	var list []Deposit
+	lastHash := ""
+
+	// Setup URL for query
+	btcURL, err := url.JoinPath(cfg.BtcExplorerURL, "address", tssAddress, "txs")
+	if err != nil {
+		return list, err
+	}
+
+	// This loop will query the bitcoin explorer for transactions associated with the TSS address. Since the api only
+	// allows a response of 25 transactions per request, several requests will be required in order to retrieve a
+	// complete list.
+	for {
+		// The Next Query is determined by the last transaction hash provided by the previous response.
+		nextQuery := btcURL
+		if lastHash != "" {
+			nextQuery, err = url.JoinPath(btcURL, "chain", lastHash)
+			if err != nil {
+				return list, err
+			}
+		}
+		// #nosec G107 url must be variable
+		res, getErr := http.Get(nextQuery)
+		if getErr != nil {
+			return list, getErr
+		}
+
+		body, readErr := ioutil.ReadAll(res.Body)
+		if readErr != nil {
+			return list, readErr
+		}
+		closeErr := res.Body.Close()
+		if closeErr != nil {
+			return list, closeErr
+		}
+
+		// NOTE: decoding json from request dynamically is not ideal, however there isn't a detailed, defined data structure
+		// provided by blockstream. Will need to create one in the future using following definition:
+		// https://github.com/Blockstream/esplora/blob/master/API.md#transaction-format
+		var txns []map[string]interface{}
+		err := json.Unmarshal(body, &txns)
+		if err != nil {
+			return list, err
+		}
+
+		if len(txns) == 0 {
+			break
+		}
+
+		fmt.Println("Length of txns: ", len(txns))
+
+		// The "/address" blockstream api provides a maximum of 25 transactions associated with a given address. This
+		// loop will iterate over that list of transactions to determine whether each transaction can be considered
+		// a deposit to ZetaChain.
+		for _, txn := range txns {
+			// Get tx hash of the current transaction
+			hash := txn["txid"].(string)
+
+			// Read the first output of the transaction and parse the destination address.
+			// This address should be the TSS address.
+			vout := txn["vout"].([]interface{})
+			vout0 := vout[0].(map[string]interface{})
+			var vout1 map[string]interface{}
+			if len(vout) > 1 {
+				vout1 = vout[1].(map[string]interface{})
+			} else {
+				continue
+			}
+			_, found := vout0["scriptpubkey"]
+			scriptpubkey := ""
+			if found {
+				scriptpubkey = vout0["scriptpubkey"].(string)
+			}
+			_, found = vout0["scriptpubkey_address"]
+			targetAddr := ""
+			if found {
+				targetAddr = vout0["scriptpubkey_address"].(string)
+			}
+
+			//Check if txn is confirmed
+			status := txn["status"].(map[string]interface{})
+			confirmed := status["confirmed"].(bool)
+			if !confirmed {
+				continue
+			}
+
+			//Filter out deposits less than min base fee
+			if vout0["value"].(float64) < 1360 {
+				continue
+			}
+
+			//Check if Deposit is a donation
+			scriptpubkey1 := vout1["scriptpubkey"].(string)
+			if len(scriptpubkey1) >= 4 && scriptpubkey1[:2] == "6a" {
```

Comment:
It feels to me the 3rd-party JSON API incurs extra tool maintenance as you'll have a maintain a JSON version of BTC intx filter which has to exactly match existing filter done by queries directly to the Bitcoin node. For instance, different type of addresses (going to have Taproot), donation filter, restricted addresses, etc.

Reply:
This was done long before the address changes, so it will need to be updated later on. Using explorers like this will limit some things but it runs much faster than parsing the actual blocks. The only thing that we need to match here is the way we are filtering in zetaclient, I will have to maintain it based on that. The address query in the explorer isn't specific, it will be a string and isn't tied to any format. 

I will just have to match what is being filtered in zetaclient.